### PR TITLE
Disallow classic mod from giving PP (for now)

### PR DESCRIPTION
**Pull request by smoogipoo**
Since we're not sure yet on the implementation of the mechanics of this mod (e.g. https://github.com/ppy/osu/issues/11769), it's easiest to just not allow it to be ranked for now. Scores coming from stable are still allowed to give PP.

### CODE REVIEW

File: osu.Server.Queues.ScoreStatisticsProcessor/Processors/ScorePerformanceProcessor.cs [hunk 1]
```diff
@@ -191,6 +191,10 @@ public static bool AllModsValidForPerformance(Mod[] mods)
 
                         continue;
 
+                    case ModClassic:
+                        // Classic mod is only allowed if it's attached to legacy scores (null build ID).
+                        return score.BuildID == null;
```
Comment (bdach) [inline osu.Server.Queues.ScoreStatisticsProcessor/Processors/ScorePerformanceProcessor.cs:196]: Isn't examining that `LegacyScoreID` is not null the standard / canonical way to check if a `SoloScoreInfo` is legacy or not? I worry about introducing multiple heuristics for the same thing.

Comment (smoogipoo) [reply osu.Server.Queues.ScoreStatisticsProcessor/Processors/ScorePerformanceProcessor.cs:196]: Hmmm...

https://github.com/ppy/osu-queue-score-statistics/blob/3def49348ed93535df38d4bc8ffdbdb94709df47/osu.Server.Queues.ScoreStatisticsProcessor/Processors/UserTotalPerformanceProcessor.cs#L119-L124

@peppy any opinions?

Comment (bdach) [reply osu.Server.Queues.ScoreStatisticsProcessor/Processors/ScorePerformanceProcessor.cs:196]: Just for context, I'm referring to

https://github.com/ppy/osu-queue-score-statistics/blob/3def49348ed93535df38d4bc8ffdbdb94709df47/osu.Server.Queues.ScoreStatisticsProcessor/ScoreStatisticsQueueProcessor.cs#L101-L105
https://github.com/ppy/osu-queue-score-statistics/blob/3def49348ed93535df38d4bc8ffdbdb94709df47/osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/DeleteImportedHighScoresCommand.cs#L78-L79

as places that use the other method.

Comment (smoogipoo) [reply osu.Server.Queues.ScoreStatisticsProcessor/Processors/ScorePerformanceProcessor.cs:196]: Will wait on a new osu! nuget package before fixing this.

**Comment by bdach** [comment]
I have attempted to fix test failures here but I am not sure everyone will agree as to what I did, so please scrutinise https://github.com/ppy/osu-queue-score-statistics/pull/180/commits/b9b51aceb9075767a91a774dbfafa2b6f852468f especially at your discretion.

Rest seems fine to me.

**Review by bdach — APPROVED** [review]

**Comment by smoogipoo** [comment]
Looks fine to me

**Review by peppy — APPROVED** [review]

**Pull request merged**